apps/openmw/mwworld/worldimp.cpp

```diff
@@ -3569,27 +3533,8 @@ namespace MWWorld
         if (ptr == getPlayerPtr() && Settings::Manager::getBool("hit fader", "GUI"))
             return;
 
-        int type = ptr.getClass().getBloodTexture(ptr);
-        std::string texture;
-        switch (type)
-        {
-        case 2:
-            texture = getFallback()->getFallbackString("Blood_Texture_2");
-            break;
-        case 1:
-            texture = getFallback()->getFallbackString("Blood_Texture_1");
-            break;
-        case 0:
-        default:
-            texture = getFallback()->getFallbackString("Blood_Texture_0");
-            break;
-        }
-
-        std::stringstream modelName;
-        modelName << "Blood_Model_";
-        int roll = Misc::Rng::rollDice(3); // [0, 2]
-        modelName << roll;
-        std::string model = "meshes\\" + getFallback()->getFallbackString(modelName.str());
+        std::string texture = getFallback()->getFallbackString("Blood_Texture_" + std::to_string(ptr.getClass().getBloodTexture(ptr)));
```

Comment:
I don't know if this is significant, but this isn't the same. Previously, any type value that wasn't 0, 1, or 2 would result in `Blood_Texture_0`. Now, for example, type 3 results in `Blood_Texture_3` and type -1 results in `Blood_Texture_-1`. Are those other values possible, and what is vanilla's behavior if so?

Reply:
It physically can't be a value other than the ones in [0; 2] range at the moment, see getBloodTexture.
Vanilla supports blood textures in [0; 7] range but there's a need of record loading tweaking to make it so in OpenMW.

Reply:
I don't see a problem, then